examples/date-picker.py

```diff
@@ -0,0 +1,113 @@
+import dash
+from dash import dcc, html
+from dash.dependencies import Input, Output, State
+import plotly.express as px
+import pandas as pd
+import numpy as np
+
+app = dash.Dash(__name__)
+
+app.layout = html.Div(
+    [
+        dcc.Store(id="stored-data", storage_type="memory"),
+        html.H1("Date Picker Example"),
+        html.H2("Step 1: Select a Date Range to generate ramdom data"),
+        html.P("Start Date"),
+        dcc.DatePickerSingle(
+            id="date-picker-single-start", clearable=True, display_format="YYYY-MM-DD"
+        ),
+        html.P("End Date"),
+        dcc.DatePickerSingle(
+            id="date-picker-single-end", clearable=True, display_format="YYYY-MM-DD"
+        ),
+        html.Div(
+            [
+                html.H2(
+                    "Step 2: Use the Date Picker Range to filter the data (within the date range) and update the graph"
+                ),
+                dcc.DatePickerRange(
+                    id="date-picker-range",
+                    display_format="YYYY-MM-DD",
+                ),
+                dcc.Graph(id="graph"),
+            ],
+            hidden=True,
+            id="graph-container",
+        ),
+    ]
+)
+
+# create a callback for when either date picker single is changed, and create a new dataframe
+@app.callback(
+    Output("date-picker-range", "start_date"),
+    Output("date-picker-range", "end_date"),
+    Output("date-picker-range", "min_date_allowed"),
+    Output("date-picker-range", "max_date_allowed"),
+    Output(
+        "graph-container", "hidden"
+    ),  # update the visibility of graph-container
+    Output("stored-data", "data"),
+    [
```

Comment:
It's no longer necessary to have the Output Input or State in a list.  This still works, but to keep the style consistent with other examples, it's better to take them out.